src/xpcc/driver/storage/block_device_heap.hpp

```diff
@@ -0,0 +1,95 @@
+// coding: utf-8
+/* Copyright (c) 2017, Raphael Lehmann
+ * All Rights Reserved.
+ *
+ * The file is part of the xpcc library and is released under the 3-clause BSD
+ * license. See the file `LICENSE` for the full license governing this code.
+ */
+// ----------------------------------------------------------------------------
+
+#ifndef XPCC_BLOCK_DEVICE_HEAP_HPP
+#define XPCC_BLOCK_DEVICE_HEAP_HPP
+
+#include <xpcc/processing/resumable.hpp>
+
+namespace xpcc
+{
+
+/**
+ * \brief	Block device using heap (RAM)
+ *
+ * Usually not persistant, but at least useful for testing
```

Comment:
This is useful for battery backed SRAM, some stm32 have up to 4kB of this stuff. Very nice!

Reply:
For this use-case `BdHeap` is now extended to optionally use externally supplied memory instead of allocation memory. [See here.](https://github.com/roboterclubaachen/xpcc/pull/328/commits/92bfb846d58c66df1e13bd8fb9c3270a510a9bb7#diff-a9fb6f431c1838169ef8fe59f70ef89fR42)